infinispan/src/main/java/org/hibernate/search/infinispan/impl/InfinispanDirectoryProvider.java

```diff
@@ -129,8 +136,30 @@ public void start(DirectoryBasedIndexManager indexManager) {
 		log.debugf( "Initialized Infinispan index: '%s'", directoryProviderName );
 	}
 
+	private void augmentDirectoryWithAsyncDeletes(org.infinispan.lucene.directory.BuildContext directoryBuildContext) {
```

Comment:
It's indeed not easy to test programatically, I've tested it building a SNAPSHOT and adding a dependency against product branches